libs/event_system/cringine/event_system/input/input_manager.hpp

```diff
@@ -1,14 +1,56 @@
 #pragma once
 
 #include <cringine/event_system/input/keys.hpp>
+#include <cringine/event_system/event_system.hpp>
+
+#include <array>
 
 namespace cringine::event_system::input
 {
-    class input_manager
+    class input_manager : virtual public events::cursor_focus_event, virtual public events::cursor_position_event,
+                          virtual public events::keyboard_input_event, virtual public events::mouse_button_input_event,
+                          virtual public events::scroll_input_event
```

Comment:
You do not need virtual inheritance here

Reply:
I think you should do something like this:
`
class input_manager : protected events::event {
...
protected:
    void event() override = 0;
...
}
`